xclim/indices/_conversion.py

```diff
@@ -904,29 +916,72 @@ def rain_approximation(
     return prra
 
 
+@declare_units(snd="[length]", snr="[mass]/[volume]", const="[mass]/[volume]")
+def snd_to_snw(
+    snd: xr.DataArray,
+    snr: Quantified | None = None,
+    const: Quantified = "312 kg m-3",
+    out_units: str = None,
+) -> xr.DataArray:
+    """Snow amount from snow depth and density.
+
+    Parameters
+    ----------
+    snd : xr.DataArray
+        Snow depth.
+    snr : Quantified, optional
+        Snow density.
+    const: Quantified
+        Constant snow density
+        `const` is only used if `snr` is None.
+    out_units: str, optional
+        Desired units of the snow amount output. If `None`, output units simply follow from `snd * snr`.
+
+    Returns
+    -------
+    xr.DataArray
+        Snow amount
+
+    Notes
+    -----
+    The estimated mean snow density value of 312 kg m-3 is taken from :cite:t:`sturm_swe_2010`.
+
+    References
+    ----------
+    :cite:cts:`sturm_swe_2010`
+    """
+    density = snr if (snr is not None) else const
+    snw = rate2flux(snd, density=density, out_units=out_units).rename("snw")
+    # TODO: Leave this operation to rate2flux? Maybe also the variable renaming above?
+    snw.attrs["standard_name"] = "surface_snow_amount"
+    return snw
+
+
 @declare_units(snw="[mass]/[area]", snr="[mass]/[volume]", const="[mass]/[volume]")
 def snw_to_snd(
     snw: xr.DataArray,
-    snr: xr.DataArray | None = None,
+    snr: Quantified | None = None,
```

Comment:
So to confirm, we only handle `Quantified` changes here from now on?

Reply:
My mistake! I was confusing this with the `Quantity` type. I didn't realize this includes `xr.DataArray`s. All good!